src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -613,6 +614,23 @@ internal static bool WriteConsideredUse(TypeSymbol type, BoundExpression value)
                 case BoundKind.ObjectCreationExpression:
                     var init = (BoundObjectCreationExpression)value;
                     return !init.Constructor.IsImplicitlyDeclared || init.InitializerExpressionOpt != null;
+                case BoundKind.ConvertedTupleLiteral:
```

Comment:
>BoundKind.ConvertedTupleLiteral [](start = 21, length = 31)

I assume we would want to use similar rules when initialization uses a tuple instance (rather than a literal) on the right. Consider adding tests for scenarios like this too.